web/src/main/java/org/akaza/openclinica/service/StudyParticipantServiceImpl.java

```diff
@@ -375,6 +409,198 @@ private boolean isEnrollmentCapEnforced(StudyBean currentStudy,StudyBean siteStu
+					        subjectContext.setStudyEventId(studyEventID);
+		
+					        subjectContext.setStudySubjectOid(studySubjectOID);
+					        studyEventOrdinal = studyEvent.getSampleOrdinal() +"";
+					        subjectContext.setOrdinal(studyEventOrdinal);
+					        subjectContext.setUserAccountId(userAccountID);
+					        UserAccountDAO udao = new UserAccountDAO(dataSource);
+					        UserAccountBean ub = (UserAccountBean) udao.findByPK(Integer.parseInt(userAccountID));
+		
+					        FormLayout formLayout = formLayoutDao.findByOcOID(subjectContext.getFormLayoutOid());
+					        Role role = Role.RESEARCHASSISTANT;
+					        String mode = PFormCache.VIEW_MODE;
+					        					
+							List<Bind> binds = openRosaServices.getBinds(formLayout,EnketoAPI.QUERY_FLAVOR,studyOID);
+					        boolean formContainsContactData=false;
+					        if(openRosaServices.isFormContainsContactData(binds))
+					            formContainsContactData=true;
+		
+					        String subjectContextKey;
+					        
+							subjectContextKey = this.createSubjectContextKey(studyOID, formLayout, studyEvent, studySubjectOID, userAccountID, servletContext);
+							File pdfFile = urlService.getFormPdf(subjectContextKey, subjectContext, studyOID, studySubjectOID,formLayout, EnketoAPI.QUERY_FLAVOR, null, role, mode, null, false,formContainsContactData,binds,ub,
+									format, margin, landscape);
+							
+							if(pdfFile !=null) {
+								pdfFiles.add(pdfFile);
+							}
+			    	    }										
+				        
+			    	}//for-loop-2	    						
+			    }//for-loop-1		   
+			    
+				mergedPdfFile = this.mergePDF(pdfFiles, studyOID, studySubjectIdentifier,fullFinalFilePathName);
+				mergedPdfFileNm = mergedPdfFile.getName();
+				userService.persistJobCompleted(jobDetail, mergedPdfFileNm);
+			} catch (Exception e) {
+	            userService.persistJobFailed(jobDetail, mergedPdfFileNm);
+	            this.writeToFile(e.getMessage(), fullFinalFilePathName);
+	            throw e;
+	        }
+		    
+			return mergedPdfFile;
+		}
+  
+    /**
+     * 
+     * @param msg
+     * @param fileName
+     */
+    public void writeToFile(String msg, String fileName) {
+        logger.debug("writing report to File");
+     
+        File file = new File(fileName);       
+
+        PrintWriter writer = null;
+        try {
+        	 file.createNewFile();
+        	 writer = new PrintWriter(file.getPath(), "UTF-8");
+        	 writer.print(msg);     
+        } catch (IOException e) {
+        	 logger.error("Error while accessing file to start writing: ",e);
+		} finally {                        
+            writer.close();;
+        }
+
+    }
+    /**
+     * AC5: The PDF Casebook will be called "Participant <Participant ID> Casebook <current timestamp>.pdf".
+     * @param files
+     * @param studyOID
+     * @param studySubjectIdentifier
+     * @return
+     * @throws IOException
+     */
+    public File mergePDF(ArrayList<File> files,
```

Comment:
move this method to a new service called "PdfService" interface and implement the interface method in "PdfServiceImpl"